packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -2980,10 +2996,15 @@ function bailoutOnAlreadyFinishedWork(
 
   // Check if the children have any pending work.
   if (!includesSomeLane(renderLanes, workInProgress.childLanes)) {
+    // TODO (effects)  The case we are trying to handle is an error thrown during commit,
+    // and we replay the work, even though we skip render phase (by bailing out)
+    // we don't want to bailout on commit effects too.
+    // However, this check seems overly aggressive in some cases.
```

Comment:
Is this comment still relevant, @lunaruan?

Reply:
Oops nope.